lightningd/peer_htlcs.c

```diff
@@ -1215,18 +1215,44 @@ static struct channel_id *calc_forwarding_channel(struct lightningd *ld,
 		return NULL;
 
 	if (p->forward_channel) {
+		log_debug(
+		    hp->channel->log,
+		    "Looking up channel by scid=%s to forward htlc_id=%" PRIu64,
+		    type_to_string(tmpctx, struct short_channel_id,
+				   p->forward_channel),
+		    hp->hin->dbid);
```

Comment:
Whoopsie daisy :+1:

Reply:
whoops! Let's make sure there's test coverage for this path then :hot_face:

Reply:
I'm pretty sure `hp` and `hp-hin` cannot be `NULL` because we're forwarding an HTLC, and if either of those were `NULL` we couldn't propagate the resolution back. I just drafted it until I a) verify that the logic holds and b) add a comment on the lines in question :-)

Reply:
Yes, "htlc_accepted" is called on incoming htlcs, to ->hin is always non-NULL.